src/bomf/loader/entityloader.py

```diff
@@ -184,22 +184,53 @@ async def load_entities(self, entities: list[_TargetEntity]) -> list[LoadingSumm
         return base_result
 
 
-_PydanticTargetModel = TypeVar("_PydanticTargetModel", bound=BaseModel)
+_PydanticTargetModel = TypeVar("_PydanticTargetModel")
+# This is not bound to BaseModel, because I didn't manage to parametrize my generics properly.
+# See https://github.com/pydantic/pydantic/issues/7345#issuecomment-1707061637 for how it _should_ work.
+# The given example works in a minimal test but not for me in the scenario below
 
 
 # pylint:disable=too-few-public-methods
-class _ListOfPydanticModels(RootModel[list[_PydanticTargetModel]], Generic[_PydanticTargetModel]):
+class _ListOfPydanticModels(RootModel[list[SerializeAsAny[_PydanticTargetModel]]], Generic[_PydanticTargetModel]):
     pass
 
 
-class PydanticJsonFileEntityLoader(JsonFileEntityLoader[_PydanticTargetModel], Generic[_PydanticTargetModel]):
+class PydanticJsonFileEntityLoader(EntityLoader[_PydanticTargetModel], Generic[_PydanticTargetModel]):
     """
     A json file entity loader specifically for pydantic models
     """
 
     def __init__(self, file_path: Path):
         """provide a file path"""
-        super().__init__(
-            file_path=file_path,
-            list_encoder=lambda x: [y.model_dump() for y in _ListOfPydanticModels(root=x).root],
-        )
+        self._file_path = file_path
+        self._file_lock = asyncio.Lock()
+
+    def _load_entries_from_file_if_exist(self) -> _ListOfPydanticModels[_PydanticTargetModel]:
+        empty_list = _ListOfPydanticModels[_PydanticTargetModel].model_validate([])
+        if not self._file_path.exists():
+            return empty_list
+        with open(self._file_path, "r", encoding="utf-8") as json_file:
+            file_body = json_file.read()
+            if not file_body:  # if the file exists but is empty
+                return empty_list
```

Comment:
kann man diese with open methode nich in eine funktion packen?
weil sonst hast du ja an zwei stellen den gleichen code mit with open logik

Reply:
mach gerne einen vorschlag. ich sehe nicht, wie wir die 2x8 zeilen sinnvoll kapseln, dabei type-safetype und lesbarkeit erhalten.